### ISSUE TITLE
Question - Version 2.0.0

### ISSUE THREAD

**Issue by VaclavElias** [vkhorikov/CSharpFunctionalExtensions#147]
Hi,

I see that version 2.0.0 was published. Do you plan to write any blog post or update documentation to address what's new/changed? Can I just update my previous version?

Would it be possible to give some summaries what was changed/updated when you publish a new version or a change log or anything suggesting improvements/changes and why to use them?

Thank you :)

**Comment by vkhorikov** [comment]
Yeah, I need to come up with a better publishing process. Simply pushing a new version is not enough.

I'll write a blog post once we settle on #148. Regarding whether you can update the version -- you can, all changes should be backward compatible, though you'll get compilation warnings and be asked to switch from `OnSuccess` and `OnBoth` to `Map`, `Tap`, and `Finally`.

**Comment by VaclavElias** [comment]
Thank you Vladimir, looking forward your blog post and thanks for answering my other questions :)

**Comment by feliperomero3** [comment]
Hi @vkhorikov 
I'm puzzled about the transition from `OnSuccess` to `Tap` naming. How is _Tap_ a closely related word to _Success_?
English is not my first language.

PS. Congratulations on your coming [Book](https://www.manning.com/books/unit-testing)! I like your writing style, clear and focused.

**Comment by vkhorikov** [comment]
Thanks much :)

`OnSuccess` has gotten too many overloads, to the point that you may get obscure compiler errors or have a result of an unexpected type somewhere in the middle of a chain of extension methods. It's been essentially split into several extensions: `Map`, `Tap`, and `Bind`. The `Tap` naming is from typescript, the other two is from functional languages such as F#.

**Comment by btull89** [comment]
Agreed I'm upgrading to 2.0 now and I don't know what a lot of these new methods are used for. I'm relying on the xml code documentation.

**Comment by space-alien** [comment]
Check out #152.

**Comment by alexmaie** [comment]
Hey, we also upgraded from 1.10 to the latest version, and we are struggling to refactor and learn the new API's.
It's not very clear yet when to use tap or map or bind.

But still, this is a great library, keep up the good work :)

**Comment by vkhorikov** [comment]
@alexmaie Yeah, the documentation is lacking currently. You should be able to convert to the correct methods if you follow the [Obsolete] attributes' recommendations, though.

**Comment by alexmaie** [comment]
Well yes and no. After doing the initial migration it was easier to grasp, but still I didn't figure out what the best practices are.
Like we should use tap when we want to perform an action. When to use Bind/Map etc.
The documentation should be more of a guide line :). Again, this is not a complain, but a constructive feedback :D.

Great Work :)
